CriticalMass/ChatNavigationButton.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  ChatNavigationButton.swift
+//  CriticalMaps
+//
+//  Created by Leonard Thomas on 4/10/19.
+//
+
+import UIKit
+
+class ChatNavigationButton: CustomButton {
+    let unreadLabel = UILabel()
+
+    init(chatManager: ChatManager) {
```

Comment:
I would argue that the Button should not know about where the number that it should display is coming from. Meaning maybe there could be a controller kind of object that sets the number. Then the button would only disply it.